lib/src/main/kotlin/com/swmansion/starknet/data/serializers/SyncPolymorphicSerializer.kt

```diff
@@ -4,16 +4,34 @@ import com.swmansion.starknet.data.types.NotSyncingResponse
 import com.swmansion.starknet.data.types.Syncing
 import com.swmansion.starknet.data.types.SyncingResponse
 import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
 import kotlinx.serialization.json.*
 
 internal object SyncPolymorphicSerializer : JsonContentPolymorphicSerializer<Syncing>(Syncing::class) {
     override fun selectDeserializer(element: JsonElement): DeserializationStrategy<Syncing> = when (element) {
-        JsonPrimitive(false) -> NotSyncingTransformingSerializer
+        JsonPrimitive(false) -> NotSyncingResponseSerializer
         else -> SyncingResponse.serializer()
     }
 }
 
-internal object NotSyncingTransformingSerializer :
-    JsonTransformingSerializer<NotSyncingResponse>(NotSyncingResponse.serializer()) {
-    override fun transformDeserialize(element: JsonElement): JsonElement = buildJsonObject { put("status", element) }
+internal object NotSyncingResponseSerializer : KSerializer<NotSyncingResponse> {
+    override val descriptor: SerialDescriptor =
+        buildClassSerialDescriptor("NotSyncingResponse")
+
+    override fun deserialize(decoder: Decoder): NotSyncingResponse {
+        require(decoder is JsonDecoder)
+        val status = decoder.decodeJsonElement().jsonPrimitive.booleanOrNull
+            ?: throw SerializationException("Invalid NotSyncingResponse value [${decoder.decodeJsonElement().jsonPrimitive}], boolean expected.")
```

Comment:
What if this value is not a primitive?

Reply:
Discussed offline: current implementation is sufficient.